scripts/run_dead_code.sh

```
@@ -0,0 +1,19 @@
+#!/bin/bash
+# Run dead code detection
+
+echo "🔍 Running dead code detection..."
+echo "================================="
+
+cd "$(dirname "$0")/../.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)